plugins/org.csstudio.trayicon/src/org/csstudio/trayicon/TrayIcon.java

```diff
@@ -0,0 +1,101 @@
+package org.csstudio.trayicon;
+
+import org.csstudio.utility.product.Activator;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Event;
+import org.eclipse.swt.widgets.Listener;
+import org.eclipse.swt.widgets.Menu;
+import org.eclipse.swt.widgets.MenuItem;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swt.widgets.TrayItem;
+import org.eclipse.ui.IWorkbenchWindow;
+import org.eclipse.ui.PlatformUI;
+
+public class TrayIcon {
+
+    public static final String IMAGE = "icons/css.ico";
+    private TrayItem trayItem;
+    final private Image image = Activator.getImageDescriptor(IMAGE).createImage();
+    private Menu menu;
+    private MenuItem open;
+    private MenuItem exit;
+    private IWorkbenchWindow window;
+
+    private boolean minimized;
+
+    private void dispose() {
+        trayItem.dispose();
```

Comment:
We're using the same image for the lifetime of the application.  We may want to do the same for the whole tray icon and just hide it or show it as appropriate.